proto/spacemesh/v1/global_state_types.proto

```diff
@@ -50,22 +50,46 @@ message AccountDataQueryRequest {
     uint32 offset = 3; // return results from offset
 }
 
+message ExecutionError {
+    uint32 code = 1; // error code
+    string message = 2; // error message
+}
+
+// From dev perspective we want to enable something like the following from the higher-level lang or rust code:
+// info([enty_name], [entry_text])
+// e.g. info("sig_verification", format!("%s : %d", str1, ui64))
+message LogEntry {
+    uint32 log_level = 1; // info, debug, trace, error, etc...
+    string context = 2; // function name e.g. withdraw()
+    string name = 3; // entry name
+    string entry = 4; // entry printed data
+}
+
 message TransactionReceipt {
     TransactionId id = 1; // the source transaction
     enum TransactionResult { // the results of STF transaction processing
         TRANSACTION_RESULT_UNSPECIFIED = 0;
         TRANSACTION_RESULT_EXECUTED = 1; // executed w/o error by the STF
         TRANSACTION_RESULT_BAD_COUNTER = 2; // unexpected transaction counter
-        TRANSACTION_RESULT_RUNTIME_EXCEPTION = 3; // app code exception
+        TRANSACTION_RESULT_RUNTIME_EXCEPTION = 3; // app code execution exception
         TRANSACTION_RESULT_INSUFFICIENT_GAS = 4; // out of gas
         TRANSACTION_RESULT_INSUFFICIENT_FUNDS = 5; // failed due to sender's insufficient funds
+        TRANSACTION_RESULT_APP_NOT_FOUND = 6; // an attempt to execute an app where there's no app at the app address
+        TRANSACTION_RESULT_FUNC_NOT_FOUND = 8; // called function not found in app
```

Comment:
is this distinguished, at the SVM level, from any other simple execution error? how/why?